iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateFeederBay.java

```diff
@@ -0,0 +1,190 @@
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.powsybl.iidm.modification.topology.ModificationReports.*;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.createNBDisconnector;
+import static com.powsybl.iidm.modification.topology.TopologyModificationUtils.createNodeBreakerSwitches;
+
+/**
+ * This method adds a new injection bay on an existing busbar section. The voltage level containing the
+ * busbar section should be described in node/breaker topology. The injection is created and connected to
+ * the busbar section with a breaker and a closed disconnector. The injection is also connected to all
+ * the parallel busbar sections, if any, with an open disconnector.
+ * @author Coline Piloquet <coline.piloquet at rte-france.com>
+ */
+public class CreateFeederBay implements NetworkModification {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateFeederBay.class);
+
+    private final InjectionAdder<?> injectionAdder;
+    private final String bbsId;
+    private final int injectionPositionOrder;
+    private final ConnectablePosition.Direction injectionDirection;
+
+    /**
+     * Constructor.
+     *
+     * @param injectionAdder         The injection adder.
+     * @param bbsId                  The ID of the existing busbar section where we want to connect the injection.
+     *                               Please note that there will be switches between this busbar section and the connection point of the injection. This switch will be closed.
+     * @param injectionPositionOrder The order of the injection to be attached from its extension {@link ConnectablePosition}.
+     * @param injectionDirection     The direction of the injection to be attached from its extension {@link ConnectablePosition}.
+     */
+    public CreateFeederBay(InjectionAdder<?> injectionAdder, String bbsId, int injectionPositionOrder, ConnectablePosition.Direction injectionDirection) {
+        this.injectionAdder = Objects.requireNonNull(injectionAdder);
+        this.bbsId = Objects.requireNonNull(bbsId);
+        this.injectionPositionOrder = injectionPositionOrder;
+        this.injectionDirection = Objects.requireNonNull(injectionDirection);
+    }
+
+    public CreateFeederBay(InjectionAdder<?> injectionAdder, String bbsId, int injectionPositionOrder) {
+        this(injectionAdder, bbsId, injectionPositionOrder, ConnectablePosition.Direction.BOTTOM);
+    }
+
+    @Override
+    public void apply(Network network, ComputationManager computationManager) {
+        apply(network);
+    }
+
+    @Override
+    public void apply(Network network) {
+        apply(network, false, Reporter.NO_OP);
+    }
+
+    /**
+     * Applies the modification to the given network. If throwException is set to true, then in case of error, an
+     * exception will be thrown. Otherwise, computation will continue but the injection will not be added to the network
+     * in case of error.
+     */
```

Comment:
This javadoc is valid for all modifications, you should move it in the interface